octobot_trading/storage/portfolio_storage.py

```diff
@@ -32,13 +36,44 @@ async def store_history(self):
             portfolio_manager.historical_portfolio_value_manager
         metadata = hist_portfolio_values_manager.get_metadata()
         # replace the whole table to ensure consistency
+        history = hist_portfolio_values_manager.get_dict_historical_values()
         await portfolio_db.upsert(commons_enums.RunDatabases.METADATA.value, metadata, None, uuid=1)
         await portfolio_db.replace_all(
             commons_enums.RunDatabases.HISTORICAL_PORTFOLIO_VALUE.value,
-            hist_portfolio_values_manager.get_dict_historical_values(),
+            history,
             cache=False
         )
         await portfolio_db.flush()
+        await self.trigger_debounced_update_auth_data()
+
+    async def _update_auth_data(self):
+        hist_portfolio_values_manager = self.exchange_manager.exchange_personal_data. \
+            portfolio_manager.historical_portfolio_value_manager
+        authenticator = authentication.Authenticator.instance()
+        history = hist_portfolio_values_manager.get_dict_historical_values()
+        if history and authenticator.is_initialized():
+            if hist_portfolio_values_manager.portfolio_manager.portfolio_value_holder.initializing_symbol_prices_pairs:
+                for symbol in hist_portfolio_values_manager.portfolio_manager.portfolio_value_holder.initializing_symbol_prices_pairs:
+                    await commons_tree.EventProvider.instance().wait_for_event(
+                        self.exchange_manager.bot_id,
+                        commons_tree.get_exchange_path(
+                            self.exchange_manager.exchange_name,
+                            commons_enums.InitializationEventExchangeTopics.PRICE.value,
+                            symbol=symbol,
+                        ),
+                        self.PRICE_INIT_TIMEOUT
+                    )
+            await authenticator.update_portfolio(
```

Comment:
Should we catch potentials errors here?

Reply:
I don't think it's necessary, it's already caught here 
![image](https://user-images.githubusercontent.com/9078616/213307580-c595ea7e-db4b-42e7-bb5b-6b54dea5edb5.png)